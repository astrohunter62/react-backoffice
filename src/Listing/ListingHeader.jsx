import React from 'react'
import PropTypes from 'prop-types'
import {
  TableCell,
  TableHead,
  TableRow,
  TableSortLabel,
} from 'material-ui/Table'
import Tooltip from 'material-ui/Tooltip'
import Checkbox from 'material-ui/Checkbox'

class ListingHeader extends React.Component {
  createSortHandler(property) {
    return (event) => {
      this.props.onRequestSort(event, property)
    }
  }

  render() {
    const {
      onSelectAllClick,
      order,
      orderBy,
      numSelected,
      rowCount,
      headers,
    } = this.props

    return (
      <TableHead>
        <TableRow>
          <TableCell padding="checkbox">
            <Checkbox
              indeterminate={numSelected > 0 && numSelected < rowCount}
              checked={numSelected === rowCount}
              onChange={onSelectAllClick}
            />
          </TableCell>

          {headers.map(column => (
            <TableCell
              key={column.id}
              padding={column.isPaddingDisabled ? 'none' : 'default'}
            >
              <Tooltip
                title="Sort"
                placement={column.numeric ? 'bottom-end' : 'bottom-start'}
                enterDelay={300}
              >
                <TableSortLabel
                  active={orderBy === column.id}
                  direction={order}
                  onClick={this.createSortHandler(column.id)}
                >
                  {column.label}
                </TableSortLabel>
              </Tooltip>
            </TableCell>
            ), this)}
        </TableRow>
      </TableHead>
    )
  }
}

ListingHeader.propTypes = {
  headers: PropTypes.arrayOf(PropTypes.object).isRequired,
  numSelected: PropTypes.number.isRequired,
  onRequestSort: PropTypes.func.isRequired,
  onSelectAllClick: PropTypes.func.isRequired,
  order: PropTypes.string.isRequired,
  orderBy: PropTypes.string.isRequired,
  rowCount: PropTypes.number.isRequired,
}

export default ListingHeader
