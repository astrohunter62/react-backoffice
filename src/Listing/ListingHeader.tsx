import React, { FunctionComponent } from "react";
import {
  Checkbox,
  TableCell,
  TableHead,
  TableRow,
  TableSortLabel,
  Tooltip,
} from "@material-ui/core";

type Props = {
  headers: Record<string, any>[];
  numSelected?: number;
  onRequestSort: (...args: any[]) => any;
  onSelectAllClick: (...args: any[]) => any;
  order?: "asc" | "desc";
  orderBy: string;
  rowCount?: number;
};

const ListingHeader: FunctionComponent<Props> = ({
  onSelectAllClick,
  order,
  orderBy,
  numSelected = 0,
  rowCount = 0,
  headers,
  onRequestSort,
}) => {
  const createSortHandler = (property: any) => (event: any) =>
    onRequestSort(event, property);

  return (
    <TableHead>
      <TableRow>
        <TableCell padding="checkbox">
          <Checkbox
            color="primary"
            indeterminate={numSelected > 0 && numSelected < rowCount}
            checked={numSelected === rowCount}
            onChange={onSelectAllClick}
          />
        </TableCell>

        {headers.map((column: any, index: number) => (
          <TableCell
            key={`header-${column.id}-${index}`}
            padding={column.isPaddingDisabled ? "none" : "default"}
            align={column.isNumeric ? "right" : undefined}
          >
            <Tooltip
              title="Sort"
              placement={column.numeric ? "bottom-end" : "bottom-start"}
              enterDelay={300}
            >
              <TableSortLabel
                active={orderBy === column.id}
                direction={order}
                onClick={createSortHandler(column.id)}
              >
                {column.label}
              </TableSortLabel>
            </Tooltip>
          </TableCell>
        ))}
      </TableRow>
    </TableHead>
  );
};

export default ListingHeader;
