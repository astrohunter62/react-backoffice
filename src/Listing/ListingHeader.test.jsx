import React from 'react'
import renderer from 'react-test-renderer'
import { Table } from 'material-ui'

import ListingHeader from './ListingHeader'

it('renders correctly', () => {
  const tree = renderer
    .create((
      <Table>
        <ListingHeader
          headers={[{
            id: 'name',
            label: 'Name',
          }]}
          numSelected={0}
          onRequestSort={() => {}}
          onSelectAllClick={() => { }}
          order="asc"
          orderBy="name"
          rowCount={0}
        />
      </Table>
    ))
    .toJSON()
  expect(tree).toMatchSnapshot()
})
