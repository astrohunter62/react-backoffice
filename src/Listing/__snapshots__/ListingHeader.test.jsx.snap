// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<table
  className="MuiTable-root-1"
>
  <thead
    className="MuiTableHead-root-2"
  >
    <tr
      className="MuiTableRow-root-3 MuiTableRow-head-6"
    >
      <th
        aria-sort={null}
        className="MuiTableCell-root-8 MuiTableCell-head-9 MuiTableCell-paddingCheckbox-14"
        scope="col"
      >
        <span
          className="MuiButtonBase-root-36 MuiIconButton-root-30 MuiPrivateSwitchBase-root-26 MuiCheckbox-root-20 MuiCheckbox-colorPrimary-24 MuiPrivateSwitchBase-checked-27 MuiCheckbox-checked-21"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <span
            className="MuiIconButton-label-35"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-39"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
              />
            </svg>
            <input
              checked={true}
              className="MuiPrivateSwitchBase-input-29"
              data-indeterminate={false}
              onChange={[Function]}
              type="checkbox"
            />
          </span>
          <span
            className="MuiTouchRipple-root-61"
          />
        </span>
      </th>
      <th
        aria-sort={null}
        className="MuiTableCell-root-8 MuiTableCell-head-9"
        scope="col"
      >
        Array [
          <span
            aria-describedby={null}
            className="MuiButtonBase-root-36 MuiTableSortLabel-root-56 MuiTableSortLabel-active-57"
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex="0"
            title="Sort"
          >
            Name
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-39 MuiTableSortLabel-icon-58 MuiTableSortLabel-iconDirectionAsc-60"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
              />
            </svg>
          </span>,
          "",
        ]
      </th>
    </tr>
  </thead>
</table>
`;
